python-main.js

```diff
@@ -488,7 +488,15 @@ function web_editor(config) {
     function generateFullHexStr() {
         var fullHexStr = '';
         try {
-            micropythonFs.write('main.py', EDITOR.getCode());
+
+            // Update or remove main.py depending on editor content
+            if(!EDITOR.getCode()) {
+                micropythonFs.remove('main.py');
```

Comment:
If the file does not exists this will throw an exception and stop processing the hex file, which we don't want, if the users clicks the download button twice we want both actions to succeed with the same output hex file.
So we need to surround this line with `if (exists('main.py') {}`.